boa/src/builtins/uri/mod.rs

```diff
@@ -0,0 +1,104 @@
+//! This module implements the global `Number` object.
+//!
+//! The `Number` JavaScript object is a wrapper object allowing you to work with numerical values.
+//! A `Number` object is created using the `Number()` constructor. A primitive type object number is created using the `Number()` **function**.
+//!
+//! The JavaScript `Number` type is double-precision 64-bit binary format IEEE 754 value. In more recent implementations,
+//! JavaScript also supports integers with arbitrary precision using the BigInt type.
+//!
+//! More information:
+//!  - [ECMAScript reference][spec]
+//!  - [MDN documentation][mdn]
+//!
+//! [spec]: https://tc39.es/ecma262/#sec-number-object
+//! [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number
+
+use super::function::make_builtin_fn;
+use std::borrow::Borrow;
+
+use crate::{
+    builtins::BuiltIn, object::ConstructorBuilder, property::Attribute, value::Value, BoaProfiler,
+    Context, Result,
+};
+
+use percent_encoding::{percent_decode, utf8_percent_encode, AsciiSet, CONTROLS};
+
+// https://url.spec.whatwg.org/#fragment-percent-encode-set
+const ENCODE_FRAGMENT: &AsciiSet = &CONTROLS.add(b' ').add(b'"').add(b'<').add(b'>').add(b'`');
+const PARSE_INT_MAX_ARG_COUNT: usize = 1;
+
+#[derive(Debug, Clone, Copy)]
+pub(crate) struct EncodeURI;
+
+impl BuiltIn for EncodeURI {
+    const NAME: &'static str = "encodeURI";
+
+    fn attribute() -> Attribute {
+        Attribute::WRITABLE | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE
+    }
+
+    fn init(context: &mut Context) -> (&'static str, Value, Attribute) {
+        let _timer = BoaProfiler::global().start_event(Self::NAME, "init");
+
+        let attribute = Attribute::READONLY | Attribute::NON_ENUMERABLE | Attribute::PERMANENT;
+        let number_object = ConstructorBuilder::with_standard_object(
+            context,
+            Self::constructor,
+            context.standard_objects().number_object().clone(),
+        )
+        .name(Self::NAME)
+        .length(Self::LENGTH)
+        .build();
+
+        let global = context.global_object();
+        make_builtin_fn(
+            Self::decode_uri,
+            "decodeURI",
```

Comment:
@Razican SHould I implement encodeURI, decodeURI etc... in separate modules? I mean - should I create 4 constructors? Or I can add all functions with `make_builtin_fn` ?

P.S. I'm not experienced Rust developer, only learning, but it is a pleasure to work with this project: easy navigation, readable and understandable code, easy to debug.

Reply:
@jasonwilliams I'm still interested in implementing this feature. Could smbd please guide me ?

Reply:
Hello @captain-yossarian, sorry for not getting back to you earlier. About this, I think It's OK having everything as you have it now. It's very reasonable. But maybe @HalidOdat can give a better response on this.

I will do a review asap.

Reply:
Hi @Razican
>sorry for not getting back to you earlier
No problem at all.
Thank you for all your comments.